gen3/bin/gitops.sh

```diff
@@ -385,13 +387,43 @@ gen3_gitops_sync() {
     portal_roll=true
   fi
 
+  if [[ ! -f "${commonsManifestDir}/etlMapping.yaml" ]]; then
+    gen3_log_info "etl mapping file not found, skipping etl update"
+  else
+    local filename
+    local configMapFile
+    local comparingFile
+    local diffMsg
+    diffMsg=""
+    filename="etlMapping.yaml"
+    commonsFilepath="${commonsManifestDir}/$filename"
+    configMapFile=$(g3kubectl get cm etl-mapping -o json | jq -r '.data."'"$filename"'"' | tr -d '\n' | base64 -w 0)
+    # get file contents from default file or commons's file
+    comparingFile=$(cat $commonsFilepath | tr -d '\n' | base64 -w 0)
+    # check for a diff
+    if [[ "${configMapFile}" = null ]]; then
+      diffMsg="Diff in etlMapping/${filename} - file not found in secret"
+    elif [[ "${configMapFile}" != "${comparingFile}" ]]; then
+      diffMsg="Diff in etlMapping/${filename} - difference between file and secret"
+    fi
+    if [[ ! -z "${diffMsg}" ]]; then
+      etlDiffs="${etlDiffs} \n${diffMsg}"
+      gen3_log_info "$diffMsg"
+      etl_roll=true
+    fi
+  fi
+
   echo "DRYRUN flag is: $GEN3_DRY_RUN"
   if [ "$GEN3_DRY_RUN" = true ]; then
     echo "DRYRUN flag detected, not rolling"
-    gen3_log_info "dict_roll: $dict_roll; versions_roll: $versions_roll; portal_roll: $portal_roll"
+    gen3_log_info "dict_roll: $dict_roll; versions_roll: $versions_roll; portal_roll: $portal_roll; etl_roll: $etl_roll"
   else
-    if [[ ( "$dict_roll" = true ) || ( "$versions_roll" = true ) || ( "$portal_roll" = true ) ]]; then
+    if [[ ( "$dict_roll" = true ) || ( "$versions_roll" = true ) || ( "$portal_roll" = true )|| ( "$etl_roll" = true ) ]]; then
       echo "changes detected, rolling"
+      # run etl job before roll all so guppy can pick up changes
+      if [[ "$etl_roll" = true ]]; then
+          gen3 job run etl
```

Comment:
I think you need:
```
gen3 update_config etl-mapping "$(gen3 gitops folder)/etlMapping.yaml"
```
before the `job run` (rebase on master to get the new `gen3 gitops folder` command :-p)

Then - you can pass `--wait` to `gen3 job run` to wait for the ETL to complete before rolling:
```
gen3 job run etl --wait
```
Pretty sure that will work.